### PR TITLE
week 5 assessment complete

### PR DESCRIPTION
**Pull request by jumba23**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ X] JavaScript Coding Challenges
- [ X] Ruby Coding Challenges
- [ X] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -10,43 +10,169 @@
...
+      "Apricot",
+      "Peach",
+    ]);
+    expect(findWords(fruitArray, letterE)).toEqual([
+      "Cherry",
+      "Blueberry",
+      "Peach",
+    ]);
+    expect(findWords(fruitArray, letterI)).toEqual(["Apricot", "Kiwi"]);
+    expect(findWords(fruitArray, letterZ)).toEqual([]);
```
Comment (elysemontano) [inline code-challenges.test.js:100]: Nice job in adding these edge cases to your tests!  👏

File: code-challenges.test.js [hunk 2]
```diff
@@ -10,43 +10,169 @@
...
+
+const findFullHouse = (cards) => {
+  let cardsFound = new Set(cards);
+  const [card1, card2] = cardsFound;
+  if (cardsFound.size != 2) return false;
+  return cards.filter((e) => e === card1).length > 3 ||
+    cards.filter((e) => e === card2).length > 3
+    ? false
+    : true;
+};
```
Comment (elysemontano) [inline code-challenges.test.js:176]: Nice work!  This is a pretty unique approach, but you did a great job documenting your process on this.  👏

File: interview-questions.md [hunk 3]
```diff
@@ -8,44 +8,44 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: Having each table with a primary key assures row-level accessibility. This allows us to access data in a more structured and predictable manner. Being able to allocate a piece of data using the primary key allows us to use CRUD operations in more efficient manner. Also, there are benefits of being able to use the primary key of one table as a foreign key in another table in effort to create associations between the two and further.
 
-Researched answer:
+Researched answer: In a relational table the primary uniquely identifies each record in the table. Databases use keys to compare, sort, and store records, and to create relationship between records. Choosing the primary key in a database is one of the most important steps in the process. It allows database operations on the table to be faster because it also serves as an index. The table's data is by default present ordered by the primary key.
```
Comment (elysemontano) [inline interview-questions.md:37]: Great work in this section!

File: ruby-challenges.rb [hunk 4]
```diff
@@ -10,21 +10,66 @@
...
+p removeNestedArrays(us_states)
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+# -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
+
 
+class Bike
```
Comment (elysemontano) [inline ruby-challenges.rb:46]: Nice work!!

File: ruby-challenges.rb [hunk 5]
```diff
@@ -10,21 +10,66 @@
...
 
+def removeNestedArrays(hash)
+    array = []
+    hash.values.each do |value|
+        value.map do |element| 
+            array << element 
+        end
+    end
+    array.sort
+end
```
Comment (elysemontano) [inline ruby-challenges.rb:35]: This looks great!  There is also a flatten method that will do some of the work for you.

**Review by elysemontano — APPROVED** [review]
Really great work Goran!  Interview answers looked good and your code logic looks excellent.  Left a few comments to look at throughout, but excellent job!  👏
